components/script/dom/document.rs

```diff
@@ -1656,6 +1668,64 @@ impl Document {
 
         self.window.layout().nodes_from_point(page_point, *client_point)
     }
+
+    // https://html.spec.whatwg.org/multipage/#sequential-focus-navigation
+    fn sequential_focus_navigation(&self, event: &Event, key: Key) {
+        // Step 1
+        // Step 2
+        // TODO: Implement the sequential focus navigation starting point.
+        // This can be used to change the starting point for navigation.
+
+        // Step 3
+        let direction =
+            if key == Key::Tab {
+                let modifier = event.downcast::<KeyboardEvent>()
+                                    .unwrap().get_key_modifiers();
+
+                if modifier.is_empty() {
+                    NavigationDirection::Forward
+                } else {
+                    NavigationDirection::Backward
+                }
+            } else {
+                return
+            };
+
+        // TODO: Step 4
+
+        // Step 5
+        let candidate = self.sequential_search(direction);
+
+        // Step 6
+        self.request_focus(&self.focus_list.borrow()[candidate]);
+    }
+
+    fn sequential_search(&self, direction: NavigationDirection) -> usize {
```

Comment:
This is supposed to return a candidate; shouldn't the return type be `Root<Element>`?